helm/frost-server/templates/_helpers.tpl

```diff
@@ -33,20 +33,174 @@ Get the HTTP service API version.
+  {{- end -}}
+{{- end -}}
+
+{{/*
+Get the default HTTP nginx rewriteAnnotations.
+*/}}
+{{- define "frost-server.http.ingressProvider.nginx.rewriteAnnotation" -}}
+  {{- $mydict := dict -}}
+  {{- if .Values.frost.http.ingress.tls.enabled -}}
+    {{- $_ := set $mydict "nginx.ingress.kubernetes.io/ssl-redirect" "true" -}}
+  {{- end -}}
+  {{- $_ := set $mydict "nginx.ingress.kubernetes.io/rewrite-target" "/FROST-Server/$1" -}}
+  {{- $mydict | toYaml -}}
+{{- end -}}
+
+{{/*
+Get the default HTTP agic rewriteAnnotations.
+*/}}
+{{- define "frost-server.http.ingressProvider.agic.rewriteAnnotation" -}}
+  {{- $mydict := dict -}}
+  {{- if .Values.frost.http.ingress.tls.enabled -}}
+    {{- $_ := set $mydict "appgw.ingress.kubernetes.io/ssl-redirect" "true" -}}
+  {{- end -}}
+  {{- $_ := set $mydict "appgw.ingress.kubernetes.io/backend-path-prefix" "/FROST-Server/" -}}
+  {{- $mydict | toYaml -}}
+{{- end -}}
+
+{{/*
+Get the default HTTP traefik rewriteAnnotations.
+*/}}
+{{- define "frost-server.http.ingressProvider.traefik.rewriteAnnotation" -}}
+  {{- $mydict := dict -}}
+  {{- if .Values.frost.http.ingress.tls.enabled -}}
+    {{- $_ := set $mydict "traefik.ingress.kubernetes.io/router.tls" "true" -}}
+  {{- end -}}
+  {{/*
+    rewrite path annotation to implement: /FROST-Server/
+  */}}
+  {{- $mydict | toYaml -}}
+{{- end -}}
+
+{{/*
+Get the default agic rewriteAnnotations for HTTP.
+*/}}
+{{- define "frost-server.http.ingress.rewriteAnnotation" -}}
+  {{- if eq .Values.frost.http.ingress.ingressProvider "agic" -}}
+    {{ template "frost-server.http.ingressProvider.agic.rewriteAnnotation" . }}
+  {{- else -}} # default to nginx
+    {{ template "frost-server.http.ingressProvider.nginx.rewriteAnnotation" . }}
+  {{- end -}}
+{{- end -}}
+
+
+{{/*
+Get the default MQTT nginx rewriteAnnotations.
+*/}}
+{{- define "frost-server.mqtt.ingressProvider.nginx.rewriteAnnotation" -}}
+  {{- $mydict := dict -}}
+  {{- if .Values.frost.mqtt.ingress.tls.enabled -}}
+    {{- $_ := set $mydict "nginx.ingress.kubernetes.io/ssl-redirect" "true" -}}
+  {{- end -}}
+  {{- $mydict | toYaml -}}
+{{- end -}}
+
+{{/*
+Get the default MQTT agic rewriteAnnotations.
+*/}}
+{{- define "frost-server.mqtt.ingressProvider.agic.rewriteAnnotation" -}}
+  {{- $mydict := dict -}}
+  {{- if .Values.frost.mqtt.ingress.tls.enabled -}}
+    {{- $_ := set $mydict "appgw.ingress.kubernetes.io/ssl-redirect" "true" -}}
+  {{- end -}}
+  {{- $mydict | toYaml -}}
+{{- end -}}
+
+{{/*
+Get the default MQTT traefik rewriteAnnotations.
+*/}}
+{{- define "frost-server.mqtt.ingressProvider.traefik.rewriteAnnotation" -}}
+  {{- $mydict := dict -}}
+  {{- if .Values.frost.mqtt.ingress.tls.enabled -}}
+    {{- $_ := set $mydict "traefik.ingress.kubernetes.io/router.tls" "true" -}}
+  {{- end -}}
+  {{- $mydict | toYaml -}}
+{{- end -}}
+
+
+{{/*
+Get the default agic rewriteAnnotations for MQTT.
```

Comment:
Shouldn't this be: "Get the default rewriteAnnotations for MQTT"? I'd suggest to move this over the specific annotation configurations.